src/Api.js

```diff
@@ -89,13 +91,100 @@ class Api {
 	}
 
 	/**
-	 * Get the WhoColor data for a given wiki page.
+	 * Extract token and editor IDs from a WikiWho span element with `id='token-X'` and
+	 * `class='token-editor-Y'` attributes.
 	 *
-	 * @param  {string} wikiUrl URL of the wiki page that we want to analyze.
+	 * @param {HTMLElement} element
+	 * @return {*[]} A two-element array: 0 = int token_id, 1 = string editor_id.
+	 */
+	getIdsFromElement( element ) {
+		const out = [ false, false ],
+			tokenMatches = element.id.match( /token-(\d+)/ ),
+			editorMatches = element.className.match( /token-editor-([^\s]+)/ );
+		if ( tokenMatches && tokenMatches[ 1 ] ) {
+			out[ 0 ] = parseInt( tokenMatches[ 1 ] );
+		}
+		if ( editorMatches && editorMatches[ 1 ] ) {
+			out[ 1 ] = editorMatches[ 1 ];
+		}
+		return out;
+	}
+
+	/**
+	 * Get the WikiWho replacement for `.mw-parser-output` HTML.
+	 * @return {string}
+	 */
+	getReplacementHtml() {
+		return this.results.extended_html;
+	}
+
+	/**
+	 * Get user and revision information for a given token.
+	 *
+	 * @param {int} tokenId
+	 * @return {{revisionId: *, score: *, userId: *, username: *, revisionTime: *}|boolean}
+	 */
+	getTokenInfo( tokenId ) {
+		let revId, revision, username, score;
+
+		// Get the token information. results.tokens structure:
+		// [ [ conflict_score, str, o_rev_id, in, out, editor/class_name, age ], ... ]
+		// e.g. Array(7) [ 0, "indicate", 769691068, [], [], "18201938", 76652371.587203 ]
+		const token = this.results.tokens[ tokenId ] || false;
+		if ( !token ) {
+			return false;
+		}
+
+		// Get revision information. results.revisions structure:
+		// { rev_id: [ timestamp, parent_rev_id, user_id, editor_name ], ... }
+		// e.g. Array(4) [ "2017-03-11T02:12:47Z", 769315355, "18201938", "Biogeographist" ]
+		revId = token[ 2 ];
+		revision = this.results.revisions[ revId ];
+		username = revision[ 3 ];
+
+		// Get the user's edit score (percentage of content edited).
+		// results.present_editors structure:
+		// [ [ username, user_id, score ], ... ]
+		for ( let i = 0; i < this.results.present_editors.length; i++ ) {
+			if ( this.results.present_editors[ i ][ 0 ] === username ) {
+				score = parseFloat( this.results.present_editors[ i ][ 2 ] ).toFixed( 1 );
+				break;
+			}
+		}
+
+		// Put it all together.
+		return {
+			username: username,
+			userId: token[ 5 ],
+			revisionId: revId,
+			revisionTime: new Date( revision[ 0 ] ),
+			score: score
+		};
+	}
+
+	/**
+	 * Get the WikiWho data for a given wiki page.
+	 *
+	 * @param {string} wikiUrl URL of the wiki page that we want to analyze.
 	 * @return {Promise}
```

Comment:
Add info about return; something like `returns a promise that resolves when the data is ready, or rejects if there was an error`

This is a bit more important since we're not returning data with the promise, it's meant to serve as a sort of flag/wait until the data's available for us to call the other class methods.